x-pack/plugins/reporting/server/routes/deprecations.ts

```diff
@@ -18,42 +19,80 @@ import { IlmPolicyManager, LevelLogger as Logger } from '../lib';
 export const registerDeprecationsRoutes = (reporting: ReportingCore, logger: Logger) => {
   const { router } = reporting.getPluginSetupDeps();
 
+  const authzWrapper = <P, Q, B>(handler: RequestHandler<P, Q, B>): RequestHandler<P, Q, B> => {
+    return async (ctx, req, res) => {
+      const {
+        core: { elasticsearch },
+      } = ctx;
+
+      const store = await reporting.getStore();
+
+      try {
+        const { body } = await elasticsearch.client.asCurrentUser.security.hasPrivileges({
+          body: {
+            index: [
+              {
+                privileges: ['manage'], // required to do anything with the reporting indices
+                names: [store.getReportingIndexPattern()],
+              },
+            ],
+          },
+        });
+
+        if (!body.has_all_requested) {
+          return res.notFound();
```

Comment:
Returng `res.unauthorized()` here created a situation where the UI would log the user out as soon as they navigated to reporting without these privileges. It looks like this is baked into how the security plugin works.